test/tests/performance/pbs_cgroups_stress.py

```diff
@@ -0,0 +1,221 @@
+        pbs_home = self.server.pbs_conf['PBS_HOME']
+        vntype_file = os.path.join(pbs_home, 'mom_priv', 'vntype')
+        self.logger.info("Deleting vntype files from moms")
+        r = self.du.rm(path=vntype_file, force=True, sudo=True, logerr=False)
+        if not r:
+            self.skipTest("pbs_cgroups_hook: need root privileges")
+
+    @staticmethod
+    def get_paths():
+        """
+        Returns a dictionary containing the location where each cgroup
+        is mounted.
+        """
+        paths = {}
+        # Loop through the mounts and collect the ones for cgroups
+        with open(os.path.join(os.sep, "proc", "mounts"), 'r') as fd:
+            for line in fd:
+                entries = line.split()
+                if entries[2] != "cgroup":
+                    continue
+                flags = entries[3].split(',')
+                subsys = os.path.basename(entries[1])
+                paths[subsys] = entries[1]
+                if 'memory' in flags:
+                    paths['memsw'] = paths['memory']
+        return paths
+
+    def load_hook(self, filename):
+        """
+        Import and enable a hook pointed to by the URL specified.
+        """
+        try:
+            with open(filename, 'r') as fd:
+                script = fd.read()
+        except IOError:
+            self.assertTrue(False, "Failed to open hook file %s" % filename)
+        events = '"execjob_begin,execjob_launch,execjob_attach,'
+        events += 'execjob_epilogue,execjob_end,exechost_startup,'
+        events += 'exechost_periodic"'
+        a = {'enabled': 'True',
+             'freq': '2',
+             'event': events}
+        self.server.create_import_hook(self.hook_name, a, script,
+                                       overwrite=True)
+        # Add the configuration
+        self.load_config(self.cfg0)
+
+    def load_config(self, cfg):
+        """
+        Create a hook configuration file with the provided contents.
+        """
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, cfg)
+        os.close(fd)
+        a = {'content-type': 'application/x-config',
+             'content-encoding': 'default',
+             'input-file': fn}
+        self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(fn)
+        self.mom.log_match('pbs_cgroups.CF;copy hook-related ' +
+                           'file request received',
+                           max_attempts=5,
+                           starttime=self.server.ctime)
+        self.logger.info("Current config: %s" % cfg)
+        # Restart MoM to work around PP-993
+        self.mom.restart()
+
+    @timeout(1800)
+    def test_cgroups_epilogue_race(self):
+        """
+        Test to verify cgroups hook does not read/write  needed files
+        """
+        pcpus = 0
+        with open('/proc/cpuinfo', 'r') as desc:
+            for line in desc:
+                if re.match('^processor', line):
+                    pcpus += 1
```

Comment:
You could do this, or just do a self.server.status() for the pcpus attribute

Reply:
This is how the current cgroups hook test checks for pcpus, I'd rather not change it.